### PR TITLE
Rearrange examples

### PR DESCRIPTION
**Pull request by vafeini**
Adds the following simpler examples:

- WalletInitiatedIssuanceUsingAuthorizationFlow
- OfferBasedIssuanceUsingAuthorizationFlow
- OfferBasedIssuanceUsingPreAuthorizationFlow

### CODE REVIEW

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/CredentialConfiguration.kt [hunk 1]
```diff
@@ -28,6 +28,13 @@ import java.util.*
  */
 sealed interface CryptographicBindingMethod : Serializable {
 
+    /**
```
Comment (babisRoutis) [inline src/main/kotlin/eu/europa/ec/eudi/openid4vci/CredentialConfiguration.kt:31]: This is not in the specification Please remove it.

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/formats/CredentialIssuerMetadataJsonParser.kt [hunk 2]
```diff
@@ -547,5 +547,6 @@ private fun cryptographicBindingMethodOf(s: String): CryptographicBindingMethod
         s == "jwk" -> CryptographicBindingMethod.JWK
         s == "cose_key" -> CryptographicBindingMethod.COSE
         s.startsWith("did") -> CryptographicBindingMethod.DID(s)
+        s == "x5c" -> CryptographicBindingMethod.X5C
```
Comment (babisRoutis) [inline src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/formats/CredentialIssuerMetadataJsonParser.kt:550]: Please remove it. Not in the specification

File: src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/Commons.kt [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+)
+
+val DefaultOpenId4VCIConfig = OpenId4VCIConfig(
+    clientId = "wallet-dev",
+    authFlowRedirectionURI = URI.create("urn:ietf:wg:oauth:2.0:oob"),
+    keyGenerationConfig = KeyGenerationConfig(Curve.P_256, 2048),
+    CredentialResponseEncryptionPolicy.SUPPORTED,
+)
+
+internal fun httpClientFactory(): HttpClient = HttpClient(Apache) {
```
Comment (babisRoutis) [inline src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/Commons.kt:56]: Consider a val instead of fun.

Why Apache engine?

File: src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/WalletInitiatedIssuanceUsingAuthorizationFlow.kt [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+import io.ktor.http.*
+import kotlinx.coroutines.runBlocking
+import org.jsoup.Jsoup
+import org.jsoup.nodes.FormElement
+import java.net.URL
+
+private val CredentialIssuer_URL = "https://dev.issuer-backend.eudiw.dev"
+private val actingUser = ActingUser("tneal", "password")
+
+fun main(): Unit = runBlocking {
```
Comment (babisRoutis) [inline src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/WalletInitiatedIssuanceUsingAuthorizationFlow.kt:32]: Consider to remain fun main to something else let's say `runUseCase` with two input parameters

- Credential Issuer Identifier &
- List of configurations id(s) 

Then provide a `main` which calls it

In this way it will be clear what is the "knowledge"  of the wallet (to initiate the flow)

File: src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/WalletInitiatedIssuanceUsingAuthorizationFlow.kt [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    val credentialConfigurationIds = listOf(PID_SdJwtVC_config_id, PID_MsoMdoc_config_id, MDL_config_id)
```
Comment (babisRoutis) [inline src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/WalletInitiatedIssuanceUsingAuthorizationFlow.kt:33]: Do we need to demonstrate 3 credentials?
We already have such an example.

Perhaps a single credential is enough

File: src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/OfferBasedIssuanceUsingAuthorizationFlow.kt [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+import io.ktor.http.*
+import kotlinx.coroutines.runBlocking
+import org.jsoup.Jsoup
+import org.jsoup.nodes.FormElement
+import java.net.URL
+
+private val actingUser = ActingUser("tneal", "password")
+
+fun main(): Unit = runBlocking {
+    val credentialOfferUrl = "eudi-openid4ci://credentialsOffer?credential_offer=%7B%22" +
```
Comment (babisRoutis) [inline src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/OfferBasedIssuanceUsingAuthorizationFlow.kt:31]: Instead of `eudi-openid4ci://credentialsOffer?...`
use `eudi-openid4ci://?...`

File: src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/OfferBasedIssuanceUsingAuthorizationFlow.kt [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+        "%22eu.europa.ec.eudiw.pid_vc_sd_jwt%22,%22org.iso.18013.5.1.mDL%22]," +
+        "%22grants%22:%7B%22authorization_code%22:%7B%22" +
+        "authorization_server%22:%22https://dev.auth.eudiw.dev/realms/pid-issuer-realm%22%7D%7D%7D"
+
+    println("[[Scenario: Issuance based on credential offer url: $credentialOfferUrl]] ")
+
+    val credentialOfferRequestResolver = CredentialOfferRequestResolver(ktorHttpClientFactory = ::httpClientFactory)
+    val credentialOffer = credentialOfferRequestResolver.resolve(credentialOfferUrl).getOrThrow()
+
+    val issuer = Issuer.make(
```
Comment (babisRoutis) [inline src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/OfferBasedIssuanceUsingAuthorizationFlow.kt:43]: Perhaps we can introduce yet another `make` (factory methody of `Issuer`)

Accepting the offered URI and the config. It would resolve the offer and invoke the current `maker`

File: src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/OfferBasedIssuanceUsingPreAuthorizationFlow.kt [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+ */
+package eu.europa.ec.eudi.openid4vci.examples
+
+import com.nimbusds.jose.jwk.Curve
+import eu.europa.ec.eudi.openid4vci.*
+import eu.europa.ec.eudi.openid4vci.internal.ensure
+import kotlinx.coroutines.runBlocking
+import java.net.URI
+
+fun main(): Unit = runBlocking {
```
Comment (babisRoutis) [inline src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/OfferBasedIssuanceUsingPreAuthorizationFlow.kt:24]: Perhaps we should move this example to another PR and re-introduce it when `x5c` binding method is clarified

**Review by babisRoutis — CHANGES_REQUESTED** [review]

File: src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/Commons.kt (hunk 3 above)
Comment (vafeini) [reply src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/Commons.kt:56]: Was added to allow integrations with servers that have self signed certificates

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/CredentialConfiguration.kt (hunk 1 above)
Comment (vafeini) [reply src/main/kotlin/eu/europa/ec/eudi/openid4vci/CredentialConfiguration.kt:31]: Done

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/formats/CredentialIssuerMetadataJsonParser.kt (hunk 2 above)
Comment (vafeini) [reply src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/formats/CredentialIssuerMetadataJsonParser.kt:550]: done

File: src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/WalletInitiatedIssuanceUsingAuthorizationFlow.kt (hunk 4 above)
Comment (vafeini) [reply src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/WalletInitiatedIssuanceUsingAuthorizationFlow.kt:32]: done

File: src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/OfferBasedIssuanceUsingAuthorizationFlow.kt (hunk 6 above)
Comment (vafeini) [reply src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/OfferBasedIssuanceUsingAuthorizationFlow.kt:31]: Done. We must update also Issuer UI to exclude it from the QR code/link generated

File: src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/OfferBasedIssuanceUsingPreAuthorizationFlow.kt (hunk 8 above)
Comment (vafeini) [reply src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/OfferBasedIssuanceUsingPreAuthorizationFlow.kt:24]: I wouyld leave it as a demonstration of how pre-authorized flow can be used. Just as a show case.

File: src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/Commons.kt (hunk 3 above)
Comment (babisRoutis) [reply src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/Commons.kt:56]: Confused by the function name.

An http factory has a type `()->HttpClient` it is a `val`. 
Alternatively, you can simply rename the existing function to `acquireHttpClient()`

**Comment by babisRoutis** [comment]
@vafeini Please rebase with main

File: src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/OfferBasedIssuanceUsingAuthorizationFlow.kt (hunk 7 above)
Comment (vafeini) [reply src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/OfferBasedIssuanceUsingAuthorizationFlow.kt:40]: done
